src/SAML2/XML/saml/BaseID.php

```diff
@@ -8,91 +8,52 @@
 use SimpleSAML\Assert\Assert;
 use SimpleSAML\SAML2\Compat\ContainerSingleton;
 use SimpleSAML\SAML2\Constants as C;
+use SimpleSAML\SAML2\Utils;
+use SimpleSAML\SAML2\XML\ExtensionPointInterface;
+use SimpleSAML\SAML2\XML\ExtensionPointTrait;
 use SimpleSAML\SAML2\XML\IDNameQualifiersTrait;
 use SimpleSAML\XML\Exception\InvalidDOMElementException;
-use SimpleSAML\XML\XMLStringElementTrait;
+use SimpleSAML\XML\Exception\SchemaViolationException;
 use SimpleSAML\XMLSecurity\Backend\EncryptionBackend;
 use SimpleSAML\XMLSecurity\XML\EncryptableElementInterface;
 use SimpleSAML\XMLSecurity\XML\EncryptableElementTrait;
 
-use function trim;
+use function count;
+use function explode;
 
 /**
  * SAML BaseID data type.
  *
  * @package simplesamlphp/saml2
  */
-class BaseID extends AbstractSamlElement implements BaseIdentifierInterface, EncryptableElementInterface
+class BaseID extends AbstractSamlElement implements
+     BaseIdentifierInterface,
+     EncryptableElementInterface,
+     ExtensionPointInterface
 {
     use EncryptableElementTrait;
+    use ExtensionPointTrait;
     use IDNameQualifiersTrait;
-    use XMLStringElementTrait;
 
     /** @var string */
     public const LOCALNAME = 'BaseID';
 
-    /** @var string */
-    protected string $type;
-
 
     /**
      * Initialize a saml:BaseID from scratch
      *
-     * @param string $type
-     * @param string $value
      * @param string|null $NameQualifier
      * @param string|null $SPNameQualifier
      */
     protected function __construct(
-        string $type,
-        string $value,
         ?string $NameQualifier = null,
         ?string $SPNameQualifier = null
     ) {
-        $this->setType($type);
-        $this->setContent($value);
         $this->setNameQualifier($NameQualifier);
         $this->setSPNameQualifier($SPNameQualifier);
     }
 
 
-    /**
-     * Get the type of this BaseID (expressed in the xsi:type attribute).
-     *
-     * @return string
-     */
-    public function getType(): string
```

Comment:
I think we need to keep this one. It's the way we have to retrieve the value we got in the `xsi:type` attribute.

Reply:
It was replaced by getXsiType on the trait, but that one you want removed... So now you got me confused